picard/util/pipe.py

```diff
@@ -104,14 +98,12 @@ def __init__(self, app_name: str, app_version: str, args=None):
         if not args:
             args = (self.MESSAGE_TO_IGNORE,)
 
-        self.__app_name: str = app_name
-        self.__app_version: str = app_version
         self.__is_mac: bool = IS_MACOS
         self.__is_win: bool = IS_WIN
 
         # named pipe values needed by windows API
         if self.__is_win:
-            self.__app_version = self.__app_version.replace(".", "-")
+            app_version = app_version.replace(".", "-")
```

Comment:
I think we should move this to `if self.__is_win` branch in `__generate_filename()`

Reply:
fair point